test/tests/functional/pbs_test_tpp.py

```diff
@@ -825,6 +829,149 @@ def test_COMM_ROUTERS_with_nondefault_pbs_conf(self):
+        f = open(out_path, "r")
+        for lines in f:
+            gdb_out += lines
+        num_wait_thread = gdb_out.count("pwait")
+        return num_wait_thread
+
+    def test_PBS_COMM_THREADS(self):
+        """
+        Test allowable values for variable PBS_COMM_THREADS of pbs.conf
+        """
+        a = {'PBS_COMM_THREADS': '2'}
+        self.set_pbs_conf(host_name=self.mom.shortname, conf_param=a)
+        comm_pid = self.comm.get_pid()
+        num_wait_thread = self.start_gdb(comm_pid)
+        _msg = "No of wait threads is not equal to 2"
+        self.assertEqual(num_wait_thread, 2, _msg)
+        a = {'PBS_COMM_THREADS': '50'}
+        self.set_pbs_conf(host_name=self.mom.shortname, conf_param=a)
+        comm_pid = self.comm.get_pid()
+        num_wait_thread = self.start_gdb(comm_pid)
+        _msg = "No of wait threads is not equal to 2"
+        self.assertEqual(num_wait_thread, 50, _msg)
+        a = {'PBS_COMM_THREADS': '100'}
+        self.set_pbs_conf(host_name=self.mom.shortname, conf_param=a)
+        comm_pid = self.comm.get_pid()
+        num_wait_thread = self.start_gdb(comm_pid)
+        _msg = "No of wait threads is not equal to 2"
+        self.assertEqual(num_wait_thread, 100, _msg)
+
+    def test_PBS_COMM_THREADS_with_nondefault_pbsconf_path(self):
+        """
+        Test allowable values for variable PBS_COMM_THREADS when pbs.conf
+        is present in non default path
+        """
+        a = {'PBS_COMM_THREADS': '2'}
+        self.set_pbs_conf(host_name=self.mom.shortname, conf_param=a)
+        self.copy_pbs_conf_to_non_default_path()
+        comm_pid = self.comm.get_pid()
+        num_wait_thread = self.start_gdb(comm_pid)
+        _msg = "No of wait threads is not equal to 2"
+        self.assertEqual(num_wait_thread, 2, _msg)
+        a = {'PBS_COMM_THREADS': '50'}
+        self.set_pbs_conf(host_name=self.mom.shortname, conf_param=a)
+        self.copy_pbs_conf_to_non_default_path()
+        comm_pid = self.comm.get_pid()
+        num_wait_thread = self.start_gdb(comm_pid)
+        _msg = "No of wait threads is not equal to 2"
+        self.assertEqual(num_wait_thread, 50, _msg)
+        a = {'PBS_COMM_THREADS': '100'}
+        self.set_pbs_conf(host_name=self.mom.shortname, conf_param=a)
+        self.copy_pbs_conf_to_non_default_path()
+        comm_pid = self.comm.get_pid()
+        num_wait_thread = self.start_gdb(comm_pid)
+        _msg = "No of wait threads is not equal to 2"
+        self.assertEqual(num_wait_thread, 100, _msg)
+
+    def common_steps_log_events(
+            self, hook_name, server_ip, existence=True, starttime=None):
+        """
+        This function contains common steps for test "PBS_COMM_LOG_EVENTS"
+        and "test_PBS_COMM_LOG_EVENTS_with_nondefault_pbsconf".
+        The steps are to verify the log msgs in comm logs whenever a hook
+        is created, hook script is imported and hook is deleted.
+        :param server_ip: Server's ip address
```

Comment:
we would need to add description for hook_name as well.
we can also remove the server_ip based on my comment below.